src/Resolver/InfiniteLoopResolver.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Setono\SyliusRedirectPlugin\Resolver;
+
+use Setono\SyliusRedirectPlugin\Model\RedirectInterface;
+use Setono\SyliusRedirectPlugin\Repository\RedirectRepositoryInterface;
+
+final class InfiniteLoopResolver implements InfiniteLoopResolverInterface
```

Comment:
Would a better name be `InfiniteLoopDetector`? I am not sure, wdyt?

Reply:
Yeah, nice name. I've struggled a bit to come with `Resolver`. `Detector` is indeed a good name.